MAUI/DataGrid/Column-types.md

```diff
@@ -521,6 +580,227 @@ dataGrid.Columns.Add(dateColumn);
+            </sfgrid:DataGridComboBoxColumn>
+        </sfgrid:SfDataGrid.Columns>
+    </sfgrid:SfDataGrid>
+{% endhighlight %}
+
+{% highlight c# %}
+public class ItemSourceSelector : IItemsSourceSelector
+{
+    public IEnumerable GetItemsSource(object record, object dataContext)
+    {
+        if (record == null)
+        {
+            return null;
+        }
+
+        var orderinfo = record as DealerInfo;
+        var countryName = orderinfo.ShipCountry;
+        var viewModel = dataContext as EditingViewModel;
+
+        // Returns ShipCity collection based on ShipCountry.
+        if (viewModel.ShipCities.ContainsKey(countryName))
+        {
+            string[] shipcities = null;
+            viewModel.ShipCities.TryGetValue(countryName, out shipcities);
+            return shipcities.ToList();
+        }
+
+        return null;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+![DataGrid with ItemSourceSelector comboBox column](Images\column-types\maui-datagrid-comboBox-column-itemsourceselector.png)
+![DataGrid with ItemSourceSelector comboBox column](Images\column-types\maui-datagrid-comboBox-column-itemsourceselector2.png)
+
+### Editing the combo box
+
+The [DataGridComboBoxColumn]() supports both editable and non-editable text box to choose selected items in given data source. Users can select one item from the suggestion list.
+
+[IsEditableMode]() property is used to enable the user input in [DataGridComboBoxColumn](). The default value is `false`.
+
+{% tabs %}
+{% highlight xaml %}
+    <local:ViewModel x:Name="viewModel" />
+    </ContentPage.BindingContext>
+
+    <sfGrid:SfDataGrid x:Name="dataGrid"
+                       ItemsSource="{Binding OrdersInfo}">
+        <sfGrid:SfDataGrid.Columns>
+            <sfgrid:DataGridComboBoxColumn BindingContext="{x:Reference viewModel}"
+                                           HeaderText="Name"
+                                           IsEditableMode="True"
+                                           ItemsSource="{Binding CustomerNames}"
+                                           MappingName="DealerName" />
+        </sfGrid:SfDataGrid.Columns>
+    </sfGrid:SfDataGrid>
+{% endhighlight %}
+
+{% highlight c# %}
+dataGrid = new SfDataGrid();
+DataGridComboBoxColumn comboBoxColumn = new DataGridComboBoxColumn()
+{
+    BindingContext = viewModel,
+    MappingName = "DealerName",
+    ItemsSource = viewModel.CustomerNames,
+    IsEditableMode = True,
+    HeaderText = "Name"
+
+};
+dataGrid.Columns.Add(comboBoxColumn);
+{% endhighlight %}
+{% endtabs %}
+
+![DataGrid with Edited comboBox column](Images\column-types\maui-datagrid-combobox-column-editing.png)
+
+### Auto suggesting on edit mode
+
+By default, auto suggestion in the dropdown will display the value based on the [StartsWith]() filter condition. This can be changed to retrieve the matches with the Contains condition by using the [SuggestionMode]() property.
```

Comment:
By default, the auto-suggestion in the dropdown will display values based on the [StartsWith]() filter condition. However, you can change this behavior by utilizing the [SuggestionMode]() property to retrieve matches using the [Contains]() condition.

Reply:
![image](https://github.com/syncfusion-content/maui-docs/assets/113881396/2f9ae9dc-c471-4113-83ca-242d86c5bac0)